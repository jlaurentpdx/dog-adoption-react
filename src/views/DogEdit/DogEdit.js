import { useEffect, useState } from 'react';
import { fetchDogById } from '../../services/dogs';

export default function DogEdit(props) {
  const id = props.match.params.id;
  const [loading, setLoading] = useState(true);
  const [dog, setDog] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) return <h1>...getting ready to introduce...</h1>;

  return <div>This is the dog edit page. You are editing {dog.name}.</div>;
}
