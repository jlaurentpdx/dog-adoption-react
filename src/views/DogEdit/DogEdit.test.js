import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import DogEdit from './DogEdit';

test('DogEdit renders a form to the page with pre-filled data', async () => {
  const { container } = render(
    <MemoryRouter>
      <DogEdit match={{ params: { id: 6 } }} />
    </MemoryRouter>
  );

  await screen.findByDisplayValue('Sadie');
  await screen.findByText(
    `When she's not busy being your admirer, Sadie likes to curl up and take a nap. Great with kids and other mellow pets!`
  );

  expect(container).toMatchSnapshot();
});
