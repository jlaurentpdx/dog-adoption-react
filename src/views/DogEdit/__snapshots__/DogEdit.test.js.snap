// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogEdit renders a form to the page with pre-filled data 1`] = `
<div>
  <div>
    <p>
      Editing 
      Sadie
    </p>
    <p />
    <form>
      <div
        class="form-control"
      >
        <label>
          Name: 
        </label>
        <input
          placeholder="enter name"
          type="text"
          value="Sadie"
        />
      </div>
      <div
        class="form-control"
      >
        <label>
          Description: 
        </label>
        <textarea
          placeholder="Who is this dog?"
          type="text"
        >
          When she's not busy being your admirer, Sadie likes to curl up and take a nap. Great with kids and other mellow pets!
        </textarea>
      </div>
      <div
        class="form-control"
      >
        <label>
          Age: 
        </label>
        <input
          placeholder="enter age in years"
          type="number"
          value="10"
        />
      </div>
      <div
        class="form-control"
      >
        <label>
          Image: 
        </label>
        <input
          placeholder="enter URL"
          type="text"
          value="https://placedog.net/400/400/6"
        />
      </div>
      <div
        class="form-control"
      >
        <label>
          Breed: 
        </label>
        <input
          placeholder="enter breed or mix"
          type="text"
          value=""
        />
      </div>
      <div
        class="form-control"
      >
        <button>
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
